lightning/src/util/config.rs

```diff
@@ -47,6 +47,8 @@ pub struct ChannelHandshakeConfig {
 	/// Default value: 1. If the value is less than 1, it is ignored and set to 1, as is required
 	/// by the protocol.
 	pub our_htlc_minimum_msat: u64,
+	/// Set to percentage of channel value used to control the channel maximum htlc value
```

Comment:
This could be a bit more specific and should include some guidance on how to set it. Something like "The maximum amount we will accept for any single incoming HTLC on our channels, expressed as a percentage of the value of the channel. Payments to or routed through us above this value will be rejected before even reaching us. If this value is too large and we are a routing node, a malicious third-party could tie up our entire channel balance in a single HTLC which they refuse to resolve quickly." This should also mention the current default value.

Reply:
Note, I don't think this comment is accurate on the last part. IIRC, "htlc_max_value_in_flight_msat" is the global inbound HTLC balance, not a single HTLC. Setting a low-percentage doesn't prevent to tie up the entire channel balance in a single HTLC. Also I think it should say _inbound liquidity_ instead of channel balance, our channel balance is the _outbound liquidity_ ?

W.r.t to the default value, I would be more generous like 15% or 20%. Let say you want to preserve liquidity equilibrium in both direction. If you reach 35/75 in one direction, you trigger a rebalancing action like a circular loop to another one of your channel. A max_htlc_value of 15% means even at 35, you still have 20% of inbound liquidity buffer to keep the channel operational, assuming the in-flight 15% settled before the rebalance operation succeed.

Or what's your thinking on how a user should selection this value ? Otherwise, if you want to limit security risk, just open a smaller channel to avoid wasting almost-never used liquidity.

Reply:
Oops! You're totally right. As to guidance, that's a much tougher question if I read the variable name correctly :).

I'm not sure its right to focus on DoS here, its more about the `our_to_self_delay` vs `cltv_expiry_delta` distinction in my mind - in my mind, the biggest thing users should know is this is the max % of their channel who's security is controlled by `cltv_expiry_delta`, the rest being secured under the `our_to_self_delay`-block-limit. Its true re-balancing is a concern, but isn't it equally a concern with multiple payments? Maybe we should put rebalancing notes in htlc_maximum_msat (which I believe we currently don't expose, but should)?

Reply:
> in my mind, the biggest thing users should know is this is the max % of their channel who's security is controlled by cltv_expiry_delta, the rest being secured under the our_to_self_delay-block-limit. 

Yep, csv_delay vs cltv expiry delta sounds a good rational for guidance. 

> Its true re-balancing is a concern, but isn't it equally a concern with multiple payments? 

I think unbalancing can be provoked as well due to multiple payments than a single one and "htlc_max_value_in_flight_msat" scope either but i would say such rebalancing concern should be noted there as we _indirectly_ set hltc_maximum_msat in function of htlc_max_value_in_flight_msat (see `get_announced_htlc_max_msat`)